packages/edit-site/src/hooks/use-global-styles-output.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockTypes, store as blocksStore } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+import { useContext, useMemo } from '@wordpress/element';
+import { generateGlobalStyles } from '@wordpress/global-styles-engine';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../store';
+import { unlock } from '../lock-unlock';
+
+const { GlobalStylesContext, useGlobalSetting } = unlock(
+	blockEditorPrivateApis
+);
+
+/**
+ * Returns the global styles output based on the provided global styles config.
+ *
+ * @param {Object}  mergedConfig       The merged global styles config.
+ * @param {boolean} disableRootPadding Disable root padding styles.
+ *
+ * @return {Array} Array of stylesheets and settings.
+ */
+export function useGlobalStylesOutputWithConfig(
+	mergedConfig = {},
+	disableRootPadding = false
+) {
+	const [ blockGap ] = useGlobalSetting( 'spacing.blockGap' );
+	const hasBlockGapSupport = blockGap !== null;
+	const hasFallbackGapSupport = ! hasBlockGapSupport;
+
+	const { disableLayoutStyles, getBlockStyles } = useSelect( ( select ) => {
+		const { getSettings } = select( editSiteStore );
+		const { getBlockStyles: getBlockStylesSelector } =
+			select( blocksStore );
+		return {
+			disableLayoutStyles: !! getSettings()?.disableLayoutStyles,
+			getBlockStyles: getBlockStylesSelector,
+		};
+	}, [] );
+
+	return useMemo( () => {
+		if ( ! mergedConfig?.styles || ! mergedConfig?.settings ) {
+			return [ [], {} ];
+		}
+
+		const blockTypes = getBlockTypes();
+
+		return generateGlobalStyles( mergedConfig, blockTypes, {
+			hasBlockGapSupport,
+			hasFallbackGapSupport,
+			disableLayoutStyles,
+			disableRootPadding,
+			getBlockStyles,
```

Comment:
Note to self: missed the object that tells the function to also generate block variation styles.

Fix here: https://github.com/WordPress/gutenberg/pull/73535